pandas/core/indexing.py

```diff
@@ -2017,17 +2056,13 @@ def _getitem_axis(self, key, axis=None):
             axis = self.axis or 0
 
         if isinstance(key, slice):
-            self._has_valid_type(key, axis)
             return self._get_slice_axis(key, axis=axis)
 
         if isinstance(key, list):
-            try:
-                key = np.asarray(key)
-            except TypeError:  # pragma: no cover
```

Comment:
this is no longer needed?

Reply:
I don't see how this was ever needed. I don't think there is _any_ way in which ``np.asarray([...])`` could raise a ``TypeError``. ``np.asarray([0, [1, 2]])`` raises ``ValueError`` (catched elsewhere).

Reply:
right my point was that this was catching *something*, so let's see what it was (and if not, of course remove)

Reply:
This was added in https://github.com/pandas-dev/pandas/pull/15504/files#diff-7489e290a23303c4db4f803011ecaf8eR1728 ... but I think it was an unnecessary precaution ( @jorisvandenbossche ?), to the best of my understanding it never catched anything.